gdk/common/CaseInsensitive.py

```diff
@@ -84,7 +84,13 @@ def _read(self, file_path):
 
     def _read_from_yaml(self, file_path: Path) -> dict:
         with open(file_path, "r", encoding="utf-8") as f:
-            return yaml.safe_load(f.read())
+            try:
+                recipe_data = yaml.safe_load(f.read())
+                return recipe_data
+            except yaml.YAMLError as err:
+                logging.error(f"Syntax error when parsing the recipe file: {file_path}")
+                utils.parse_yaml_error(err)
+                sys.exit(1)
```

Comment:
raise Exception here with the error log message you have.

Reply:
Using raise Exception would trigger a lengthy traceback, which might not be the most user-friendly. Given that we can already pinpoint that the exception is due to a recipe syntax error and provide clear information about the problematic file, do we still need to raise the exception in this context?

Reply:
Yes, we're moving towards printing the full stack trace with the recent versions of GDK. Here, you can raise exception without creating a new one and keep it consistent with how rest of the GDK works.